### PR TITLE
Add prettier

### LINKED ISSUE #10: add prettier to the codebase
- today prettier isn't in the codebase, this is an issue because formatting differs between developers. 

1.) add a prettier format to the codebase
2.) add some form of a post commit/test that we can use as part of the PR process

### PR DESCRIPTION
**Pull request by alex-grover**
* Add prettier + eslint-config-prettier to turn off conflicting rules
* Add lint-staged and husky to run linter and formatter on changed files pre-commit
* Add GitHub Actions config to run linter and formatter in CI
* Run formatter and manually fix some eslint warnings

Closes #10

### CODE REVIEW

**Comment by alex-grover** [comment]
cc @MrKevinOConnell

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
```
Comment (kevoconnell) [inline .github/workflows/ci.yml]: this is so sweet lets go

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (kevoconnell) [inline .prettierrc]: probably good to have something here imo

```suggestion
{
  "printWidth": 80,
  "tabWidth": 2,
  "useTabs": false,
  "semi": true,
  "singleQuote": true,
  "trailingComma": "es5",
  "bracketSpacing": true,
  "jsxBracketSameLine": false,
  "arrowParens": "always",
  "endOfLine": "lf"
}
```

**Review by kevoconnell — CHANGES_REQUESTED** [review]
this will be so helpful thank you so much, just one suggestion (happy to have other settings - this is mostly a nit)

**Comment by kevoconnell** [comment]
> this will be so helpful thank you so much, just one suggestion (happy to have other settings - this is mostly a nit)

cc @alex-grover mb should have mentioned you

File: .prettierrc (hunk 2 above)
Comment (alex-grover) [reply .prettierrc]: up for whatever your preferences are, I included singleQuote and trailingComma from the above as the rest of those are the default settings!

**Review by kevoconnell — APPROVED** [review]
lgtm

**Pull request merged**